lib/nrf_modem_lib/trace_medium.c

```diff
@@ -0,0 +1,115 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <stddef.h>
+#include <string.h>
+#include <zephyr.h>
+#include <nrf_modem.h>
+#include <modem/trace_medium.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(trace_medium, CONFIG_NRF_MODEM_LIB_LOG_LEVEL);
+
+static struct trace_medium_entry *active_medium;
+static struct trace_medium_entry *new_medium;
+static bool switch_active_medium;
+
+static void select_first_ready_medium(void)
+{
+	struct trace_medium_entry *medium = NULL;
+	const char *medium_name = "none";
+
+	STRUCT_SECTION_FOREACH(trace_medium_entry, e) {
```

Comment:
I suggest only allowing one trace medium at compile time. That should reduce complexity in both finding the ready medium and `trace_medium_select` can be removed entirely.